sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1220,6 +1220,40 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
           Entry->second.initialize(TypeSize, DeviceImageScopeDecorated);
         }
       }
+      // ... and initialize associated host_pipe information
```

Comment:
Just a note to not forget: it seems like we should refactor this code a bit to have this function clean and tidy so splitting it to multiple functions seems like a good idea.

Reply:
I created an issue here: https://github.com/intel/llvm/issues/5938

There will need to be 3 functions to wrap